static/app/components/events/interfaces/frame/stacktraceLinkModal.tsx

```diff
@@ -1,58 +1,55 @@
-import {Component, Fragment} from 'react';
+import {Fragment, useCallback, useState} from 'react';
 import styled from '@emotion/styled';
 
 import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
 import {ModalRenderProps} from 'sentry/actionCreators/modal';
-import {Client} from 'sentry/api';
 import Alert from 'sentry/components/alert';
 import Button from 'sentry/components/button';
-import ButtonBar from 'sentry/components/buttonBar';
 import InputField from 'sentry/components/forms/inputField';
 import {t, tct} from 'sentry/locale';
 import space from 'sentry/styles/space';
-import {Integration, Organization, Project} from 'sentry/types';
+import type {Integration, Organization, Project} from 'sentry/types';
 import {
   getIntegrationIcon,
   trackIntegrationAnalytics,
 } from 'sentry/utils/integrationUtil';
-import withApi from 'sentry/utils/withApi';
+import useApi from 'sentry/utils/useApi';
 
-type Props = ModalRenderProps & {
-  api: Client;
+interface StacktraceLinkModalProps extends ModalRenderProps {
   filename: string;
   integrations: Integration[];
   onSubmit: () => void;
   organization: Organization;
   project: Project;
-};
-
-type State = {
-  sourceCodeInput: string;
-};
+}
 
-class StacktraceLinkModal extends Component<Props, State> {
-  state: State = {
-    sourceCodeInput: '',
+function StacktraceLinkModal({
+  closeModal,
+  onSubmit,
+  organization,
+  integrations,
+  filename,
+  project,
+  Header,
+  Body,
+}: StacktraceLinkModalProps) {
+  const api = useApi();
+  const [sourceCodeInput, setSourceCodeInput] = useState('');
+
+  const onHandleChange = (input: string) => {
+    setSourceCodeInput(input);
   };
 
-  onHandleChange(sourceCodeInput: string) {
-    this.setState({
-      sourceCodeInput,
-    });
-  }
-
-  onManualSetup(provider: string) {
+  function onManualSetup(provider: string) {
     trackIntegrationAnalytics('integrations.stacktrace_manual_option_clicked', {
       view: 'stacktrace_issue_details',
       setup_type: 'manual',
       provider,
-      organization: this.props.organization,
+      organization,
     });
   }
 
-  handleSubmit = async () => {
-    const {sourceCodeInput} = this.state;
-    const {api, closeModal, filename, onSubmit, organization, project} = this.props;
+  const handleSubmit = useCallback(async () => {
```

Comment:
are you sure we should add `async` here?

Reply:
we probably don't need useCallback